PluginFramework/src/Formatter/BaseFormatter.cs

```diff
@@ -0,0 +1,103 @@
+﻿using PluginFramework.src.DataContainer;
+using PluginFramework.src.Enums;
+using PluginFramework.src.EventMessages;
+using PluginFramework.src.Interfaces.PluginTypes;
+using System;
+using System.IO;
+
+namespace PluginFramework.src.Formatter
+{
+    /// <summary>
+    /// Base class for formatting
+    /// </summary>
+    public abstract class BaseFormatter : IFormatter
+    {
+        /// <inheritdoc/>
+        public string Extension { get; }
+
+        /// <inheritdoc/>
+        public PluginInformation Information { get; }
+
+        /// <inheritdoc/>
+        public event EventHandler<BaseEventArgs> StatusChanged;
+
+        protected BaseFormatter(string fileExtension, PluginInformation pluginInformation)
+        {
+            Extension = fileExtension;
+            Information = pluginInformation;
+        }
+
+        /// <inheritdoc/>
+        public bool ConvertToFormat(string filePath, string outputName, ModesEnum mode)
+        {
+            switch (mode)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5494671)